staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -249,6 +251,30 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 		exporter = nil
 	}
 
+	// Take the resetFields provided by the update strategy. If it doesn't provide them, check create (for resources that don't implement update)
+	// TODO(kwiesmueller): check if this is necessary
+	var resetFields rest.ResetFields
+	if a.group.OpenAPIModels != nil && utilfeature.DefaultFeatureGate.Enabled(features.ServerSideApply) {
```

Comment:
If we decide that we use the same logic for create and update, then we should be able to drastically simplify this logic. I'm not a big fan of `UpdateStrategyGetter` and the `Preparator` interfaces. I suspect we should be able to keep it down to 1 new interface?